.github/workflows/ecr-docker-build.yml

```diff
@@ -0,0 +1,50 @@
+name: Docker build and push to ECR
+on:
+  push:
+    branches:
+      - main
+jobs:
+  dockerBuildAndPush:
+    name: Docker build and push
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    environment: test
+    steps:
+      - uses: actions/checkout@dcd71f646680f2efd8db4afa5ad64fdcba30e748
+        with:
+          fetch-depth: '0'
+      - name: Set up AWS creds
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.ACTIONS_ROLE_ARN }}
+          aws-region: eu-west-2
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@aaf69d68aa3fb14c1d5a6be9ac61fe15b48453a2
+      - name: Create tag
+        id: create-tag
+        run: |
+          IMAGE_TAG="${{ github.sha }}-$(date +'%Y-%m-%d-%H%M%S')"
+          echo "::set-output name=image_tag::$IMAGE_TAG"
+      - name: Build docker image
+        env:
+          IMAGE_TAG: ${{ steps.create-tag.outputs.image_tag }}
+        run: |
+          cd ${GITHUB_WORKSPACE}
```

Comment:
Just ran the bits of shell through `shellcheck` and we should use the following whenever we attempt to `cd`, but make it `|| exit 1` so the job fails.

```
cd ${GITHUB_WORKSPACE}
^--------------------^ SC2164 (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```

It also points out some places where we reference variables without quoting to prevent splitting, so this line should become...

```
cd "${GITHUB_WORKSPACE}" || exit 1
```

Reply:
good shout, done